app/src/main/java/com/sakal/playlistmaker/player/ui/activity/AudioPlayerActivity.kt

```diff
@@ -19,39 +21,59 @@ class AudioPlayerActivity : AppCompatActivity() {
 
     private lateinit var binding: ActivityAudioplayerBinding
     private val viewModel by viewModel<AudioPlayerViewModel>()
-    private lateinit var track: Track
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityAudioplayerBinding.inflate(layoutInflater)
         setContentView(binding.root)
 
-        track = viewModel.getTrack()
+        @Suppress("DEPRECATION")
+        val track = intent.getSerializableExtra(Constants.TRACK) as Track
 
         viewModel.observeState().observe(this) {
             render(it)
         }
 
+        viewModel.observeFavoriteState().observe(this) {
+            renderLikeButton(it)
+        }
+
         initToolbar()
 
         showTrack(track)
 
+        viewModel.isFavorite(track.trackId)
```

Comment:
🍏 Я бы рекомендовала из активити не давать вьюмодели четких указаний по тому, что надо сделать. 
Можно просто передать ей полученную информацию, а она сама решит, что с не сделать.
Например,
```suggestion
        viewModel.onNewTrack(track.trackId)
```
или 

```suggestion
        viewModel.init(track.trackId)
```
А вьмодель под капотом уже вызовет метод isFavorite